Subject: Replace fts_accpath with fts_path in restore.c.
Author: Philip Tricca <flihp@twobit.us>

To label filesystems in an alternative root directory we need to use the
'-r' flag with the setfiles utility. The current implementation uses the
fts_accpath member of the FTSENT structure which is reletive to the "current
directory". This fails in the pseudo environment but for a reason I don't
fully understand.

This patch replaces the use of fts_accpath with fts_path which is relative
to the root specified in the traversal. By invoking setfiles with an absolute
path we're ensured that fts_path will be an absolute path to the file being
relabeled.

Index: policycoreutils/setfiles/setfiles.c
===================================================================
--- policycoreutils.orig/setfiles/setfiles.c
+++ policycoreutils/setfiles/setfiles.c
@@ -194,7 +194,7 @@ int main(int argc, char **argv)
 		r_opts.expand_realpath = 0;
 		r_opts.abort_on_error = 1;
 		r_opts.add_assoc = 1;
-		r_opts.fts_flags = FTS_PHYSICAL | FTS_XDEV;
+		r_opts.fts_flags = FTS_PHYSICAL | FTS_XDEV | FTS_NOCHDIR;
 		ctx_validate = 1;
 	} else {
 		/*
